### ISSUE TITLE
Doesn't Change the Taskbar

### ISSUE THREAD

**Issue by m4l0n** [torchgm/RoundedTB#20]
I use another TaskBar mod known as TaskBarX as well. I tried changing the margin and radius to 2 6, but it did not work. The taskbar remains the same. Reinstalled multiple times too.

**Comment by torchgm** [comment]
If you're using TranslucentTB or TaskbarX's options for custom blurs/opacity etc. then the taskbar doesn't refresh its appearance automatically and it can appear to have no effect. This is a known bug with Windows' Desktop Window Manager (DWM) and in the next release I hope to have some kind of mitigation for it.

If you close TaskbarX and make sure it's not applying any custom appearance to the taskbar, does RoundedTB work then?

**Comment by m4l0n** [comment]
Yeah, I am applying the blue.opacity effects on my Taskbar X. It works after I close Taskbar X. Apparently the rounded corners are visible when I hover over the Windows icon of the bottom left.

**Comment by torchgm** [comment]
Yeah, this is the issue with DWM I mentioned. This isn't the fault of TaskbarX or RoundedTB, just a quirk of the desktop window manager. I'm working to find the best possible mitigation for this issue. In current releases though, I'd recommend using [TranslucentTB](https://github.com/TranslucentTB/TranslucentTB) for the transparency effects (you can still use TaskbarX for centring and such). Then, from TranslucentTB, bind one of the events to the "normal" effect: 

![image](https://user-images.githubusercontent.com/31840547/126810264-0cbc9f7f-6399-4be8-a58e-68f1bec57f86.png)

In this example I'm using "start menu opened", which will mean the taskbar will go back to its normal appearance when the start menu is opened. This subsequently causes DWM to realise the taskbar has been rounded, and redraws it properly. This way, if RoundedTB ever gets "stuck", it'll fix itself as soon as you open the start menu.

Also as a side-note, TranslucentTB offers more customisation over the translucency, colour and status of the taskbar than TBX does, so if you're into customisation I'd recommend you use it anyway.

**Comment by m4l0n** [comment]
Yeah thanks. Your suggestion worked well for me.

**Comment by m4l0n** [comment]
> Yeah, this is the issue with DWM I mentioned. This isn't the fault of TaskbarX or RoundedTB, just a quirk of the desktop window manager. I'm working to find the best possible mitigation for this issue. In current releases though, I'd recommend using [TranslucentTB](https://github.com/TranslucentTB/TranslucentTB) for the transparency effects (you can still use TaskbarX for centring and such). Then, from TranslucentTB, bind one of the events to the "normal" effect:
> 
> ![image](https://user-images.githubusercontent.com/31840547/126810264-0cbc9f7f-6399-4be8-a58e-68f1bec57f86.png)
> 
> In this example I'm using "start menu opened", which will mean the taskbar will go back to its normal appearance when the start menu is opened. This subsequently causes DWM to realise the taskbar has been rounded, and redraws it properly. This way, if RoundedTB ever gets "stuck", it'll fix itself as soon as you open the start menu.
> 
> Also as a side-note, TranslucentTB offers more customisation over the translucency, colour and status of the taskbar than TBX does, so if you're into customisation I'd recommend you use it anyway.

Out of topic, but may I know which application did you use to customize that context menu?

**Comment by torchgm** [comment]
This isn't customised, it's just how it appears in Windows 11.
